Pipeline/SimplePackageDeploy/SimplePackageDeploy.groovy

```diff
@@ -0,0 +1,276 @@
+def targetLibLLQMap = parseJSONStringToMap(props.targetLibLLQMap)
+
+// Define and create tar file extract directory 
+def tarExtractDirName = "${props.workDir}/DeployFiles_${props.startTime}"
+def tarExtractDir = new File("${tarExtractDirName}")
+
+!tarExtractDir.exists() ?: tarExtractDir.deleteDir()
+tarExtractDir.mkdirs()
+println("** Created tar file extract directory $tarExtractDirName")       
+
+// Check if the tar file provided is existing 
+def tarFile = new File("${props.tarFileName}")
+
+if(!tarFile.exists()){
+    println("** Tar file not found at $tarFile")
+    System.exit(1)
+}
+
+
+// Untar the tar file
+println("** Untar file at $tarFile.")
+
+def processCmd = ["sh", "-c", "tar -C $tarExtractDirName -xvf $tarFile"]
+
+def rc = runProcess(processCmd)
+if (rc == 0) println("Package untar done to $tarExtractDirName")
+else {
+	println("Failed to untar $tarFile")
+	System.exit(1)
+}
+
+
+// Check if BuildReport.json exists, deploy files if exists and if not stop deploy 
+
+def buildReportFile = new File("${tarExtractDirName}/BuildReport.json")
+if (buildReportFile.exists()) {	
+	deployFromBuildReport(buildReportFile,tarExtractDirName,targetLibLLQMap,copyModeMap)
+} else {
+	println("** Build report data at $tarExtractDirName/BuildReport.json not found")
+	println("** Deployment stopped")
+}
+
+if (tarExtractDir.exists()) tarExtractDir.deleteDir()
+println("\nCleaning up the temporary folder - ${tarExtractDirName}\n")
+
+/**********************************************************************************
+ **** Deploy from the BuildReport                  
+ **********************************************************************************/
+def deployFromBuildReport (File buildReportFile, String tarExtractDirName,Map targetLibLLQMap,Map copyModeMap) {
+	
+	println("\n** Deploying the contents in ${buildReportFile}")
+	
+	def buildReport= BuildReport.parse(new FileInputStream(buildReportFile))
+	def tarExtractDir = new File("${tarExtractDirName}")
+
+	//finds all the build outputs with a deployType
+	def executes= buildReport.getRecords().findAll{
+	 try {
+	     (it.getType()==DefaultRecordFactory.TYPE_EXECUTE &&
+	             !it.getOutputs().isEmpty())
+	 } catch (Exception e){}
+	}
+
+
+	//New CopyToPDS instance 
+	CopyToPDS copy = new CopyToPDS()
+
+	//Copy the extracted files to the corresponding target PDS
+	executes.each { 
+	 it.getOutputs().each {
+	     if (it.deployType != null) {
+	         def (dataset, member) = it.dataset.split("\\(|\\)")
+	         def srcFilePath = tarExtractDirName + "/" + dataset + "/" + member		
+	         if (props.packageWithExtension && props.packageWithExtension.toBoolean()) {	
+	         	srcFilePath = srcFilePath + ".${it.deployType}"
+	         }
+	         
+	         def targetPDS = "${props.hlq}" + "." + targetLibLLQMap["${it.deployType}"]
+	         println("\nExtracted file $srcFilePath is of type ${it.deployType}") 
```

Comment:
I don't think that we need this extra output.

The next output line should be sufficient by the reader.

Reply:
Deleted extra output.